packages/dev/core/src/Misc/equirectangularCapture.ts

```diff
@@ -0,0 +1,100 @@
+
+    /**
+     * Optional argument to specify position in 3D Space from where the equirectangular capture should be taken, if not specified, it would take the position of the scene's active camera or else origin
+     */
+    position?: Vector3;
+}
+
+/**
+ * @param scene This refers to the scene which would be rendered in the given equirectangular capture
+ * @param options This refers to the options for a given equirectangular capture
+ * @returns the requested capture's pixel-data or auto downloads the file if options.filename is specified
+ */
+export async function captureEquirectangularFromScene(scene: Scene, options: EquiRectangularCaptureOptions): Promise<ArrayBufferView | null> {
+    const probe = new ReflectionProbe("tempProbe", options.size, scene);
+    if (options.position) {
+        probe.position = options.position.clone();
+    } else if (scene.activeCamera) {
+        probe.position = scene.activeCamera.position.clone();
+    }
+    const meshesToConsider = options.meshesFilter ? scene.meshes.filter(options.meshesFilter) : scene.meshes;
+    probe.renderList?.push(...meshesToConsider);
+    probe.refreshRate = RenderTargetTexture.REFRESHRATE_RENDER_ONCE;
+    probe.cubeTexture.render();
+    const dumpTexture = new CustomProceduralTexture("tempProceduralTexture", "EquirectangularPanorama", { width: options.size * 2, height: options.size }, scene);
+    dumpTexture.setTexture("cubeMap", probe.cubeTexture);
+    return new Promise((resolve, reject) => {
+        dumpTexture.onGeneratedObservable.addOnce(() => {
+            const pixelDataPromise = dumpTexture.readPixels();
+            if (!pixelDataPromise) {
+                reject(new Error("No Pixel Data found on procedural texture"));
+                dumpTexture.dispose();
+                probe.dispose();
+                return;
+            }
+            pixelDataPromise.then((pixelData) => {
+                dumpTexture.dispose();
+                probe.dispose();
+                if (options.filename) {
+                    DumpTools.DumpData(options.size * 2, options.size, pixelData, undefined, "image/png", options.filename);
+                    resolve(null);
+                } else {
+                    resolve(pixelData);
+                }
+            });
+        });
+    });
+}
+
+ShaderStore.ShadersStore["EquirectangularPanoramaPixelShader"] = `
+    #ifdef GL_ES
+    precision highp float;
+    #endif
+
+    #define M_PI 3.1415926535897932384626433832795
+
+    varying vec2 vUV;
+    uniform samplerCube cubeMap;
+    void main(void) {
+
+        vec2 uv = vUV;
+        float longitude = uv.x * 2. * M_PI - M_PI + M_PI / 2.;
+        float latitude = (1. - uv.y) * M_PI;
+        vec3 dir = vec3(
+            - sin( longitude ) * sin( latitude ),
+            cos( latitude ),
+            - cos( longitude ) * sin( latitude )
+        );
+        
+        normalize( dir );
+
+        gl_FragColor = textureCube( cubeMap, dir );
+    }
+`;

```

Comment:
Would be cleaner to move this to the shader directory. You can then import the shader code here and it will be automatically populated (and minified a bit...)

Reply:
this is done on the latest commit